src/btree/bt_compact.c

```diff
@@ -113,7 +113,7 @@ __compact_page_replace_addr(WT_SESSION_IMPL *session, WT_REF *ref, WT_ADDR_COPY
     WT_ASSERT(session, addr != NULL);
 
     if (__wt_off_page(ref->home, addr))
-        __wt_free(session, addr->addr);
+        __wt_addr_address_safe_free(session, addr);
```

Comment:
Is the code in the `err` path safe:
```
err:
    if (addr != ref->addr)
        __wt_free(session, addr);
    return (ret);
```

Reply:
Yes. In case of any error, we are freeing the allocated `addr` at line 120.